syntaxes/haskell.YAML-tmLanguage

```diff
@@ -610,7 +640,7 @@ repository:
           (?x)
           ([\p{Ll}_][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}']*)
           (?:\s*,\s*([\p{Ll}_][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}']*))*
-          \s+(::|∷)
+          \s+(::|∷)(?![\p{S}\p{P}&&[^(),;\[\]`{}_"'']])
```

Comment:
This (and below) fixes the issue with `a ::: b` being recognised as a type annotation.

Reply:
Ah, interesting. Not a case I had ever thought about. :+1: